agimus_controller/mpc.py

```diff
@@ -1,380 +1,326 @@
+
+        # Filling the OCP dictionnary used to store all the results
+        self._ocp_params = {}
+        self._ocp_params["N_h"] = self._T
+        self._ocp_params["dt"] = self._dt
+        self._ocp_params["maxiter"] = self._max_iter
+        self._ocp_params["pin_model"] = robot_simulator.pin_robot.model
+        self._ocp_params["armature"] = self._OCP._runningModel.differential.armature
+        self._ocp_params["id_endeff"] = robot_simulator.pin_robot.model.getFrameId(
+            "panda2_leftfinger"
         )
-
-    def get_state_residual(self, xref):
-        return crocoddyl.CostModelResidual(
-            self.state,  # x_reg_weights,
-            crocoddyl.ResidualModelState(self.state, xref, self.actuation.nu),
+        self._ocp_params["active_costs"] = self._sqp.problem.runningModels[
+            0
+        ].differential.costs.active.tolist()
+
+        # Filling the simu parameters dictionnary
+        self._sim_params = {}
+        self._sim_params["sim_freq"] = int(1.0 / env.dt)
+        self._sim_params["mpc_freq"] = 1000
+        self._sim_params["T_sim"] = T_sim
+        self._log_rate = 100
+
+        # Initialize simulation data
+        self._sim_data = mpc_utils.init_sim_data(
+            self._sim_params, self._ocp_params, self._x0
         )
 
-    def get_xlimit_residual(self):
-        """Return velocity residual of desired joint."""
-        return crocoddyl.CostModelResidual(
-            self.state,
-            crocoddyl.ActivationModelQuadraticBarrier(
-                crocoddyl.ActivationBounds(self.state.lb, self.state.ub)
-            ),
-            crocoddyl.ResidualModelState(
-                self.state,
-                np.array([0] * (self.nq + self.nv)),
-                self.actuation.nu,
-            ),
+        # Display target in pybullet
+        mpc_utils.display_ball(
+            self._TARGET_POSE_1.translation, RADIUS=0.05, COLOR=[1.0, 0.0, 0.0, 0.6]
         )
-
-    def get_translation_residual(self):
-        """Return translation residual to the last position of the sub path."""
-        q_final = self.x_plan[-1, : self.nq]
-        target = self.robot.placement(q_final, self.nq)
-        return crocoddyl.ResidualModelFrameTranslation(
-            self.state,
-            self.robot.model.getFrameId(self.last_joint_name),
-            target.translation,
+        mpc_utils.display_ball(
+            self._TARGET_POSE_2.translation, RADIUS=0.5e-1, COLOR=[1.0, 0.0, 0.0, 0.6]
         )
 
-    def create_whole_problem(self):
-        x0 = self.x_plan[0, :]
-
-        self.whole_problem = crocoddyl.ShootingProblem(
-            x0, self.running_models, self.terminal_model
-        )
-        self.whole_problem_models = list(self.whole_problem.runningModels)
+        # Boolean describing whether the MPC has already been solved or not
+        self._SOLVE = False
+
+    def solve(self):
+        """Solve the MPC problem and display it in a pybullet GUI."""
```

Comment:
We need a different API here at some point.
We need a way to call `mpc.solve(t)` at each iteration.
The `for` loop below should `call solve(t)`
Of course the input of the solve function is at you discretion.